controllers/dependencies.go

```diff
@@ -27,10 +26,18 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.O
 	// yaml manifests that need to be applied beforehand
 	// ref: https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/ci-pipeline.md#step-6---install-the-certification-pipeline-and-dependencies-into-the-cluster
 
+	// calling remove first so that we always clone the repo in case reconciliation happens, but defer has yet to execute
+	// if the directory does not exist os.RemoveAll returns nil so there is no need to catch exceptions here
+	_ = r.removePipelineDependencyFiles(REPO_CLONE_PATH)
+
 	_, err := git.PlainClone(REPO_CLONE_PATH, false, &git.CloneOptions{
 		URL: OPERATOR_PIPELINES_REPO,
 	})
-	defer r.removePipelineDependencyFiles(REPO_CLONE_PATH)
+
+	defer func(r *OperatorPipelineReconciler, filePath string) {
+		_ = r.removePipelineDependencyFiles(filePath)
+	}(r, REPO_CLONE_PATH)
```

Comment:
I would like to have others's opinion on this too: I believe we should create a temporary dir and clone the repo into that dir. Using a non-temp dir might cause surprises if the defer method fails or does not execute for any reason.

Reply:
I'd agree with this. I would also say that having this be a member of the struct is probably overkill? This seems overly complicated. If it's a tmpdir, you could just 'defer os.RemoveAll(tmpdir)' and be done with it.

Reply:
@samira-barouti @bcrochet Please take a look at latest changes. I think this covers your concerns.